lib/core/src/Cardano/Wallet/Primitive/Types.hs

```diff
@@ -876,6 +877,13 @@ slotStartTime epochLength (SlotLength slotLength) (StartTime start) sl =
   where
     offset = slotLength * fromIntegral (flatSlot epochLength sl)
 
+-- | The SlotId at a given time.
+slotAtTime :: EpochLength -> SlotLength -> StartTime -> UTCTime -> SlotId
+slotAtTime epochLength (SlotLength slotLength) (StartTime start) time =
+    fromFlatSlot
+        epochLength
+        (round $ (time `diffUTCTime` start) / slotLength)
```

Comment:
No sure about the `round` here :thinking:  ... Shouldn't it be `floor` ?  Like, if we have let's say a slotlength or 20s and, 32 seconds, we should be in slot 1, not 2. Though, this uses `fromFlatSlot` which also does another round of conversions. 

What about computing the slot and the epoch directly from the time ?

```hs
t = time `diffUTCTime` start
epochDuration = epochLength * slotLength
epochNumber = floor (t / epochDuration)
slotNumber = floor ((t - epochNumber * epochDuration) / slotLength) 
```

I think... ?

Reply:
Note that I am pretty sure we already do something similar in the restoration benchmark code.